libs/cardano-ledger-core/src/Cardano/Ledger/TxIn.hs

```diff
@@ -52,7 +52,7 @@ import NoThunks.Class (NoThunks (..))
 -- ====================================================================================
 
 -- | A unique ID of a transaction, which is computable from the transaction.
-newtype TxId c = TxId {_unTxId :: SafeHash c EraIndependentTxBody}
```

Comment:
given that this one might be used by sever downstream libraries, would it makes sense to make a standalone `_unTxId` function with the same semantics, and immediately deprecate it?

Reply:
We can't really create a function with exact semantics as a record field has. In other words a function cannot be used to for updates or construction, eg:
```haskell
let txId = TxId { _unTxId = "...."}
```

However, adding at least a deprecated function `_unTxId` will not hurt. Done.

Reply:
of course, that makes sense, thanks! what you added is exactly what I had in mind.